File: Course_1-Algorithmic_Toolbox/Week-3/Excercise_Challenges/6_maximum_number_of_prizes/different_summands.cpp
#include <iostream>
#include <vector>
#define PB push_back
using namespace std;

vector<int> optimal_summands(int n)
{
    vector<int> summands;
    for(int i = 1; i <= n; ++i){
        n -= i;

        if(n == 0 || n <= i){
            summands.PB(n + i);
            break;
        }
        else{
            summands.PB(i);
        }
    }

    return summands;
}

int main(void)
{
    int n;
    cin >> n;
  
    vector<int> summands = optimal_summands(n);
    cout << summands.size() << '\n';
    for(size_t i = 0; i < summands.size(); ++i){
        cout << summands[i] << ' ';
    }

    return 0;
}

File: Course_1-Algorithmic_Toolbox/Week-3/Excercise_Challenges/6_maximum_number_of_prizes/different_summands.py
def optimal_summands(n):
    summands = []
    for i in range(1, n+1):
        n -= i
        if n == 0 or n <= i:
            summands.append(n + i)
            break
        else:
            summands.append(i)

    return summands

if __name__ == '__main__':
    n = int(input())
    summands = optimal_summands(n)
    print(len(summands))
    for x in summands:
        print(x, end=' ')
